pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -169,8 +174,91 @@ pub mod pallet {
 	impl<T: Config> Pallet<T> {
 		#[pallet::call_index(0)]
 		#[pallet::weight(Weight::from_parts(10_000, 0) + T::DbWeight::get().writes(1))]
-		pub fn create_pool(_origin: OriginFor<T>) -> DispatchResult {
-			todo!()
+		pub fn create_pool(
+			origin: OriginFor<T>,
+			curve: CurveInput<CurveParameterInputOf<T>>,
+			currencies: BoundedVec<TokenMetaOf<T>, T::MaxCurrencies>,
+			state: PoolStatus<Locks>,
+			denomination: u8,
+			pool_manager: AccountIdOf<T>,
+			transferable: bool,
+			team: Team<AccountIdOf<T>>,
+		) -> DispatchResult {
+			let who = T::PoolCreateOrigin::ensure_origin(origin)?;
+
+			let currency_length = currencies.len();
```

Comment:
If you take a `BoundedVec` in input, it is already guaranteed to be the max length, so you don't need to perform these checks. Different is, if you take a regular `Vec` as input, which is not recommended anyway.

Reply:
Fair point: [b01e5f6](https://github.com/KILTprotocol/kilt-node/pull/768/commits/b01e5f6fa289a6c99c1d96f120301b818b76c576)